pkg/services/ngalert/provisioning/notification_policies.go

```diff
@@ -169,3 +170,23 @@ func (nps *NotificationPolicyService) receiversToMap(records []*definitions.Post
 	}
 	return receivers, nil
 }
+
+func (nps *NotificationPolicyService) ensureDefaultReceiverExists(cfg *definitions.PostableUserConfig, defaultCfg *definitions.PostableUserConfig) {
+	defaultRcv := cfg.AlertmanagerConfig.Route.Receiver
+
+	for _, rcv := range cfg.AlertmanagerConfig.Receivers {
+		if rcv.Name == defaultRcv {
+			return
+		}
+	}
+
+	for _, rcv := range defaultCfg.AlertmanagerConfig.Receivers {
+		if rcv.Name == defaultRcv {
+			cfg.AlertmanagerConfig.Receivers = append(cfg.AlertmanagerConfig.Receivers, rcv)
+			return
+		}
+	}
+
+	nps.log.Error("Grafana Alerting has been configured with a default configuration that is internally inconsistent! The default configuration's notification policy must have a corresponding receiver.")
+	panic("inconsistent default config")
```

Comment:
I'm not sure if we should panic here. Perhaps we should instead return an error and propagate it up the call stack?

Reply:
Sure - done. I originally went with panic because having a broken default config will make things go REALLY wrong and it's not recoverable. Less of an error to be returned to the user, and more of an overall "Grafana is badly configured." But you're right, the stack trace really isn't important here and we are logging it anyway.